tests/jest/__snapshots__/enzyme-intl.test.jsx.snap

```diff
@@ -0,0 +1,245 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`mountWithIntl using FormattedMessage should match the snapshot 1`] = `
+<FormattedMessageSubject
+  foo="bar"
+  intl={
+    Object {
+      "defaultFormats": Object {},
+      "defaultLocale": "en",
+      "formatDate": [Function],
+      "formatHTMLMessage": [Function],
+      "formatMessage": [Function],
+      "formatNumber": [Function],
+      "formatPlural": [Function],
+      "formatRelative": [Function],
+      "formatTime": [Function],
+      "formats": Object {},
+      "formatters": Object {
+        "getDateTimeFormat": [Function],
+        "getMessageFormat": [Function],
+        "getNumberFormat": [Function],
+        "getPluralFormat": [Function],
+        "getRelativeFormat": [Function],
+      },
+      "locale": "en",
+      "messages": null,
+      "now": [Function],
+      "onError": [Function],
+      "textComponent": "span",
+      "timeZone": null,
+    }
+  }
+>
+  <div>
+    <FormattedMessage
+      id="TerraEnzymeIntl.helloWorld"
+      values={Object {}}
+    >
+      <span>
+        TerraEnzymeIntl.helloWorld
+      </span>
+    </FormattedMessage>
+    <FormattedMessage
+      id="TerraEnzymeIntl.buttonText"
```

Comment:
If we are mocking intl such that no translations messages are used and the fallback translation key is always shown, is it possible to mock the entire intl object for cleaner snapshots?

For example, this [dumb wrapper injects the intl context](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/tests/jest/Select.test.jsx#L11) and doesn't clutter the snapshot: https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/tests/jest/__snapshots__/Select.test.jsx.snap#L4

Reply:
We could expose the contexts (similar to what has currently been [done](https://github.com/cerner/terra-framework/blob/1de3386f4f788ca8399cac82f772f9d251f46d16/packages/terra-notification-dialog/tests/jest/intl-context-setup.js)) as well and you could pass them in as the second argument to `shallow`, `mount`, etc.

Reply:
It would essentially just be replacing the `messages` object here https://github.com/cerner/terra-framework/blob/1de3386f4f788ca8399cac82f772f9d251f46d16/packages/terra-notification-dialog/tests/jest/intl-context-setup.js#L3 with the message proxy

```diff
import { IntlProvider, intlShape } from 'react-intl';
-import messages from '../../translations/en-US.json';
+const messages = new Proxy({}, { get: (_, property) => property });

const locale = 'en';
const intlProvider = new IntlProvider({ locale, messages }, {});
const { intl } = intlProvider.getChildContext();

export const shallowContext = { context: { intl } };
export const mountContext = { context: { intl }, childContextTypes: { intl: intlShape } };
```

Reply:
I thought about this more and realized that the `terra-select` [isn't wrapped in `injectIntl`](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/Select.jsx#L231) (or using `<FormattedMessage />`) so it won't be wrapped in any extra "noise" or "clutter"

I'll add an example similar to the `terra-select` you linked

Reply:
So, I added the following example:

```jsx
const ContextTypesSubject = (props, { intl }) => (
  <div>
    {intl.formatMessage({ id: 'TerraEnzymeIntl.helloWorld' })}
    <button type="button">{intl.formatMessage({ id: 'TerraEnzymeIntl.buttonText' })}</button>
    {prettyPrintProps(props)}
  </div>
);
ContextTypesSubject.contextTypes = {
  /* eslint-disable consistent-return */
  intl: (context) => {
    if (context.intl === undefined) {
      return new Error('Component is internationalized, and must be wrapped in terra-base');
    }
  },
  /* eslint-enable consistent-return */
};

describe('shallowWithIntl', () => {
  describe('using contextTypes', () => {
    const subject = <ContextTypesSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
```

which produced the following snapshot:

```js
exports[`shallowWithIntl using contextTypes should match the snapshot 1`] = `
<div>
  TerraEnzymeIntl.helloWorld
  <button
    type="button"
  >
    TerraEnzymeIntl.buttonText
  </button>
  {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
</div>
`;
```

The `intl` object is added in via the method `const intlNode = node => cloneElement(node, { intl });` in the helpers file.

I'm not opposed to extending the helpers to include the exporting of the shallow and mount contexts so that you can do something like `shallow(Component, shallowContext)` which would remove the `intl` object from the snapshot above.

Reply:
See https://github.com/cerner/terra-enzyme-intl/pull/1/commits/de924cb9760cd1fd0a64e7e067d8f81281439b05 for enhanced test cases and the tests of using the shallow/mount contexts